### PR TITLE
Add custom panic hook for RPC port bind

### LINKED ISSUE #1005: testnet-sanity panic: solana-jsonrpchttp.worker3 panic: Address already in use
#960 appears to have caused an issue with testnet-sanity.  See https://buildkite.com/solana-labs/testnet-sanity/builds/871#33874f3a-91d2-4ccf-b033-522ca0883939

```
INFO 2018-08-18T22:28:42.715169657Z: solana::fullnode: creating bank...
--
  | INFO 2018-08-18T22:28:44.214224018Z: solana::fullnode: processing ledger...
  | INFO 2018-08-18T22:28:49.156724865Z: solana::fullnode: processed 4321 ledger...
  | INFO 2018-08-18T22:28:49.156766150Z: solana::fullnode: creating networking stack...
  | INFO 2018-08-18T22:28:49.156777967Z: solana::fullnode: starting... local gossip address: 0.0.0.0:9001 (advertising 10.138.15.237:9001)
  | thread 'thread 'solana-jsonrpchttp.worker3' panicked at '' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 98, kind: AddrInUse, message: "Address already in use" }Server initialization awaits local address.: "SendError(..)"', ', libcore/result.rslibcore/result.rs::945945::55
```

In the mean time, I've set
```
NO_VALIDATOR_SANITY=1
```
to keep us going.  Ref: https://github.com/solana-labs/solana/blob/master/ci/testnet-sanity.sh#L54

### PR DESCRIPTION
**Pull request by CriesofCarrots**
Fixes #1005 

If JSON-RPC service encounters an AddrInUse error, it prints the error, but does not panic.

I imagine that the direction we'll want to move eventually is to allow users to customize the RPC port of their node.

### CODE REVIEW

File: src/rpc.rs [hunk 1]
```diff
@@ -28,6 +29,10 @@ impl JsonRpcService {
                 let rpc = RpcSolImpl;
                 io.extend_with(rpc.to_delegate());
 
+                panic::set_hook(Box::new(move |_| {
```
Comment (mvines) [inline src/rpc.rs]: Wait but why do we panic in the first place?  Rather than a blind `unwrap()` somewhere could we just check for error?

Comment (CriesofCarrots) [reply src/rpc.rs]: Fair question. We can do that for the unwrap in the solana rpc module (which is the panic in testnet-sanity build 959). But there are actually 2 potential panics here; the 2nd one (which appears in build 871) is in a thread spawned within the jsonrpc crate, which we do not touch.
The panic::set_hook was my attempt to handle both at once.

If it seems worthwhile, I could dig into the jsonrpc crate and make a contribution there instead.

Comment (mvines) [reply src/rpc.rs]: Interesting.  Yeah we could split this into two:  (1) a fix for our unwrap, and (2) let’s try to get that panic removed/cleaned-up upstream  — if there’s no response we can look at alternatives but hopefully it won’t get to that

**Comment by CriesofCarrots** [comment]
@mvines , just pushed an error catch for (1) our unwrap.

Will start looking at (2) the upstream issue more closely, and try to resolve. It doesn't seem to panic every time, incidentally.

**Comment by mvines** [comment]
662ba4c2e3c6792042c71e6e0a1aff92181cd273 lgtm

**Comment by mvines** [comment]
Probably wanna squash before merging though!

**Comment by mvines** [comment]
Heh owell

**Comment by CriesofCarrots** [comment]
Bugger

**Pull request merged**